src/UnitTests/PubnubApi.Tests/EncryptionTests.cs

```diff
@@ -794,46 +814,76 @@ public void TestPAMv3Signature()
         }
 
         [Test]
-        async public void TestSubscribeDecryption()
+        public void TestSubscribeDecryption()
         {
-            bool done = false;
+            server.ClearRequests();
+            server.Start();
+
+            ManualResetEvent done = new ManualResetEvent(false);
             PNConfiguration config = CreateTestConfig();
+            config.LogVerbosity = PNLogVerbosity.BODY;
             config.CryptoModule = new CryptoModule(new AesCbcCryptor("enigma"), new List<ICryptor> { new LegacyCryptor("enigma") });
 
-            Pubnub sut = new Pubnub(config);
-
-            sut.AddListener(new SubscribeCallbackExt(
+            Pubnub pn = createPubNubInstance(config);
+            pn.AddListener(new SubscribeCallbackExt(
                         (pb, message) =>
                         {
                             Assert.AreEqual("test", message.Message);
-                            done = true;
+                            done.Set();
                         },
                         (pb, presence) => {},
                         (pb, status) => {}
                     )
             );
 
-            sut.Subscribe<string>().Channels(new[] { "test" }).Execute();
-
-            Pubnub sender = new Pubnub(CreateTestConfig());
-
-            // Rust generated encrypted message
-            await sender.Publish()
-                .Channel("test")
-                .Message("UE5FRAFBQ1JIEALf+E65kseYJwTw2J6BUk9MePHiCcBCS+8ykXLkBIOA")
-                .ExecuteAsync();
-
-            // It will wait until the message is received or unit test timeout is reached
-            while (!done)
-            {
-                await System.Threading.Tasks.Task.Delay(100);
-            }
+            // Time call because subscribe loop uses it before makine subscribe call
```

Comment:
@jakub-grzesiowski : FYI, This is first attempt to migrate test which was running on live server to mock one.

At Run  Unit Test CI- This is first failure.